aten/src/THNN/generic/MultiLabelMarginCriterion.c

```diff
@@ -176,10 +226,33 @@ void THNN_(MultiLabelMarginCriterion_updateGradInput)(
     isTarget_data += dim;
     gradInput_data += dim;
   }
+  gradInput_data -= nframe*dim;
```

Comment:
You should do `gradInput_data = THTensor_(data)(gradInput)` instead so the code is more readable.

Could you assert that THTensor_(isContiguous)(gradInput) is true? The following code works under this assumption and I'm not 100% sure gradInput will always be contiguous, but I'll get back to you on that.